tests/sles4sap/ipaddr2/deploy.pm

```diff
@@ -0,0 +1,38 @@
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+# Summary: Create a VM with a single NIC and 3 ip-config
+# Maintainer: QE-SAP <qe-sap@suse.de>, Michele Pagot <michele.pagot@suse.com>
+
+use strict;
+use warnings;
+use Mojo::Base 'publiccloud::basetest';
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use sles4sap::ipaddr2;
+
+sub run {
+    my ($self) = @_;
+
+    die('Azure is the only CSP supported for the moment')
+      unless check_var('PUBLIC_CLOUD_PROVIDER', 'AZURE');
+
+    select_serial_terminal;
+
+    # Init all the PC gears (ssh keys, CSP credentials)
+    my $provider = $self->provider_factory();
+
+    ipaddr2_azure_deployment(region => $provider->provider_client->region, os => get_required_var('CLUSTER_OS_VER'));
```

Comment:
I am leaving it up to your decision, but I think the content of this function could be directly part of this test module. 
This would allow to see directly deployment steps instead of just function name when you open the test module from OpenQA result page.

Reply:
it is like this as doing so it is "testable" in a UT. It means one more degree of indirection for developers inspecting the test module code. But it means I can validate more and more code in UT before to run VR in openQA

Reply:
Yeah I see where you are coming from and yes it sucks we don't have means to UT test modules. However there is the other side of the coin which is test review. In case of test failure, if reviewer wants to check what the module does, all they see is only this:

```
sub run {
    my ($self) = @_;

    die('Azure is the only CSP supported for the moment')
      unless check_var('PUBLIC_CLOUD_PROVIDER', 'AZURE');

    select_serial_terminal;

    # Init all the PC gears (ssh keys, CSP credentials)
    my $provider = $self->provider_factory();

    ipaddr2_azure_deployment(region => $provider->provider_client->region, os => get_required_var('CLUSTER_OS_VER'));
}
```

In order to check the actual code they will need to open github page (or local git repo), find which library `ipaddr2_azure_deployment` comes from and then review the code. In comparison to clicking on module name and see something directly in openQA portal. 
Yes I agree that UT is nice thing to have but there needs to be some balance. This approach makes developers life easier but at the expense of taking a tool from reviewers hands.

This might be a good topic for upcoming workshop.